controllers/dynamic_resource_request_controller.go

```diff
@@ -224,6 +182,70 @@ func (r *dynamicResourceRequestController) deleteResources(ctx context.Context,
 	return fastRequeue, nil
 }
 
+func (r *dynamicResourceRequestController) getDeletePipelinePod(ctx context.Context, resourceRequestIdentifier string, logger logr.Logger) (*v1.Pod, error) {
+	pods, err := r.getPodsWithLabels(pipeline.DeletePipelineLabels(resourceRequestIdentifier, r.promiseIdentifier))
+	if err != nil {
+		return nil, err
+	}
+	if len(pods) == 0 {
+		return nil, nil
+	}
+	return &pods[0], nil
+}
+
+func (r *dynamicResourceRequestController) configurePipelinePodHasBeenCreated(resourceRequestIdentifier string) bool {
+	pods, err := r.getPodsWithLabels(pipeline.ConfigurePipelineLabels(resourceRequestIdentifier, r.promiseIdentifier))
+	if err != nil {
+		return false
```

Comment:
should we assume that an error here means the pod hasn't been created? 🤔

Reply:
good call, should probably return an error